docs/source/perf/spmd_basic.md

```diff
@@ -61,56 +65,133 @@ SPMD with other distributed libraries.
+  `xr.global_runtime_device_attributes()`.
+  The simplest ordering is `np.array(range(num_devices))`, but tuning the
+  device ordering and the mesh shape to utilize the underlying physical
+  interconnect can improve efficiency.
+- Users are also encouraged to give each mesh axis a name. Later we can
+  shard each dimension of tensors over specific mesh axes to achieve the
+  desired parallelization. In the above example, the first mesh dimension is the
+  `data` dimension and the second mesh dimension is the `model` dimension.
 
 You can also check more mesh info via
 
 ``` python
     >>> mesh.shape()
     OrderedDict([('data', 4), ('model', 1)])
+
+    >>> mesh.get_logical_mesh()
+    array([[0], [1], [2], [3]])
+
+    # Details about these 4 TPUs
+    >>> xr.global_runtime_device_attributes()
+    [{'core_on_chip': 0, 'num_cores': 1, 'coords': [0, 0, 0], 'name': 'TPU:0'},
+     {'core_on_chip': 0, 'num_cores': 1, 'coords': [1, 0, 0], 'name': 'TPU:1'},
+     {'core_on_chip': 0, 'num_cores': 1, 'coords': [0, 1, 0], 'name': 'TPU:2'},
+     {'core_on_chip': 0, 'num_cores': 1, 'coords': [1, 1, 0], 'name': 'TPU:3'}]
+```
+
+If your workload is run on [multiple TPU slices][multislice] simultaneously, the
+device attributes will include a `slice_index` indicating the containing slice:
+
+``` python
+    # Details about 8 TPUs allocated over 2 slices
+    >>> xr.global_runtime_device_attributes()
+    [{'num_cores': 1, 'core_on_chip': 0, 'slice_index': 0, 'coords': [0, 0, 0], 'name': 'TPU:0'},
+     {'num_cores': 1, 'core_on_chip': 0, 'slice_index': 0, 'coords': [1, 0, 0], 'name': 'TPU:1'},
+     {'num_cores': 1, 'core_on_chip': 0, 'slice_index': 0, 'coords': [0, 1, 0], 'name': 'TPU:2'},
+     {'num_cores': 1, 'core_on_chip': 0, 'slice_index': 0, 'coords': [1, 1, 0], 'name': 'TPU:3'},
+     {'num_cores': 1, 'core_on_chip': 0, 'slice_index': 1, 'coords': [0, 0, 0], 'name': 'TPU:4'},
+     {'num_cores': 1, 'core_on_chip': 0, 'slice_index': 1, 'coords': [1, 0, 0], 'name': 'TPU:5'},
+     {'num_cores': 1, 'core_on_chip': 0, 'slice_index': 1, 'coords': [0, 1, 0], 'name': 'TPU:6'},
+     {'num_cores': 1, 'core_on_chip': 0, 'slice_index': 1, 'coords': [1, 1, 0], 'name': 'TPU:7'}]
 ```
 
-### Partition Spec
+In this example, a device ID of `7` will refer to the TPU at coordinate
+`[1, 1, 0]` in the second slice.
 
-partition_spec has the same rank as the input tensor. Each dimension
+### Partition spec
+
+In the following snippet:
+
+```python
+partition_spec = ('data', 'model')
+xs.mark_sharding(t, mesh, partition_spec)
+```
+
+`partition_spec` has the same rank as the input tensor. Each dimension
 describes how the corresponding input tensor dimension is sharded across
 the device mesh. In the above example tensor `t`'s fist dimension is
-being sharded at `data` dimension and the second dimension is being
-sharded at `model` dimension.
+being sharded over the `data` dimension and the second dimension is being
```

Comment:
In the above example, the first dimension of tensor `t` is sharded over the `data` dimension and the second dimension is sharded over the `model` dimension.

Reply:
done